### PR TITLE
[new OS] Alpine OptoElectronics TDCM-EDFA support

### PR DESCRIPTION
**Pull request by charlyforot**
* Add Alpine OptoElectronics mib dir
* Add new OS Alpine OptoElectronics TDCM-EDFA platform support
* Add discovery .yaml file for this OS
* Add Alpine OptoElectronics logo
* Add Alpine OptoElectronics icon
* Add tests .snmprec and .json

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/definitions/alpineoe-tdcmedfa.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    - { graph: device_bits, text: 'Device Traffic' }
+    - { graph: device_processor, text: 'CPU Usage' }
+    - { graph: device_mempool, text: 'Memory Usage' }
+discovery:
+    - 
+        sysObjectID:
+            - .1.3.6.1.4.1.52326.
+    -
+        sysDescr:
+            - One-U System
```
Comment (Jellyfrog) [inline includes/definitions/alpineoe-tdcmedfa.yaml:16]: is this needed? isnt the sysobjectid check enough?

**Comment by laf** [comment]
Any links to the vendor for these devices?

**Comment by charlyforot** [comment]
> Any links to the vendor for these devices?

Here is the link : https://www.alpineoptoelectronics.com/?section=21&module=productcatelogmodule&src=5a1c7f981609f&tse=764q6p17u3&action=view_product&id=25&cid=777&subtheme=_none

File: includes/definitions/alpineoe-tdcmedfa.yaml (hunk 1 above)
Comment (charlyforot) [reply includes/definitions/alpineoe-tdcmedfa.yaml:16]: Unfortunately not, when we remove the sysDescr field, the device is discovered as "Generic Device"

Comment (Jellyfrog) [reply includes/definitions/alpineoe-tdcmedfa.yaml:16]: because of the dot in the end I guess

Comment (charlyforot) [reply includes/definitions/alpineoe-tdcmedfa.yaml:16]: That's right, I'm going to correct this.

Comment (charlyforot) [reply includes/definitions/alpineoe-tdcmedfa.yaml:16]: That's right, I'm going to remove the sysDescr field.

**Comment by laf** [comment]
> > Any links to the vendor for these devices?
> 
> Here is the link : https://www.alpineoptoelectronics.com/?section=21&module=productcatelogmodule&src=5a1c7f981609f&tse=764q6p17u3&action=view_product&id=25&cid=777&subtheme=_none

Awesome thanks.

Is their absolutely no name for the OS/Firmware these things run?

**Comment by charlyforot** [comment]
> > > Any links to the vendor for these devices?
> > 
> > 
> > Here is the link : https://www.alpineoptoelectronics.com/?section=21&module=productcatelogmodule&src=5a1c7f981609f&tse=764q6p17u3&action=view_product&id=25&cid=777&subtheme=_none
> 
> Awesome thanks.
> 
> Is their absolutely no name for the OS/Firmware these things run?


Your're welcome.
Nothing found about an OS/firmware name, the vendor says that is a program running on a microcontroller.

**Comment by charlyforot** [comment]
@Jellyfrog Do I have to change something else ?

**Review by Jellyfrog — APPROVED** [review]

**Pull request merged**